File: Algorithms/145. Binary Tree Postorder Traversal/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public List<Integer> postorderTraversal(TreeNode root) {
    	List<Integer> list=new ArrayList<Integer>();
        return postorder(root,list);
    }
    public List<Integer> postorder(TreeNode root,List<Integer> list){
    	if (root==null) {
			return list;
		}
    	list=postorder(root.left,list);
    	list=postorder(root.right, list);
    	list.add(root.val);
    	return list;
    }
}

File: Algorithms/145. Binary Tree Postorder Traversal/Solution.py
#coding=utf-8
__author__ = 'xuxuan'
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def postorderTraversal(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        if root is None:
            return []
        ans=[]
        ans.extend(self.postorderTraversal(root.left))
        ans.extend(self.postorderTraversal(root.right))
        ans.append(root.val)
        return ans
